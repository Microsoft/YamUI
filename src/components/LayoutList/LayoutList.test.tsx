/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { LayoutList, LayoutListProps, GutterSize } from '.';

describe('<LayoutList />', () => {
  let component: ShallowWrapper<LayoutListProps, {}>;

  describe('as horizontal list', () => {
    describe('without props', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="horizontal" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with align "right"', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="horizontal" align="right" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with gutterSize', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="horizontal" gutterSize={GutterSize.LARGE} />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with additional className', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="horizontal" className="TEST_CLASSNAME" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with child content', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="horizontal">list content</LayoutList>);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('as vertical list', () => {
    describe('without props', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="vertical" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with additional className', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="vertical" className="TEST_CLASSNAME" />);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('with child content', () => {
      beforeEach(() => {
        component = shallow(<LayoutList direction="vertical">list content</LayoutList>);
      });

      it('matches its snapshot', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });
});
