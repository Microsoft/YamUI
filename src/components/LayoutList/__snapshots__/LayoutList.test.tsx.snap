// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutList /> as horizontal list with additional className matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      TEST_CLASSNAME
      {
        display: flex;
        flex-flow: row wrap;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0.4rem;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-right: 0px;
      }
/>
`;

exports[`<LayoutList /> as horizontal list with align "right" matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: flex;
        flex-flow: row wrap;
        justify-content: flex-end;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0.4rem;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-right: 0px;
      }
/>
`;

exports[`<LayoutList /> as horizontal list with child content matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: flex;
        flex-flow: row wrap;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0.4rem;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-right: 0px;
      }
>
  list content
</ul>
`;

exports[`<LayoutList /> as horizontal list with gutterSize matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: flex;
        flex-flow: row wrap;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 1.6rem;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-right: 0px;
      }
/>
`;

exports[`<LayoutList /> as horizontal list without props matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: flex;
        flex-flow: row wrap;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0.4rem;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-right: 0px;
      }
/>
`;

exports[`<LayoutList /> as vertical list with LayoutListItem child content matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: block;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        display: block;
        margin-bottom: 0;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-bottom: 0px;
      }
>
  <LayoutListItem
    className="TEST_CLASSNAME"
  />
</ul>
`;

exports[`<LayoutList /> as vertical list with additional className matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      TEST_CLASSNAME
      {
        display: block;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        display: block;
        margin-bottom: 0;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-bottom: 0px;
      }
/>
`;

exports[`<LayoutList /> as vertical list with child content matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: block;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        display: block;
        margin-bottom: 0;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-bottom: 0px;
      }
>
  list content
</ul>
`;

exports[`<LayoutList /> as vertical list with gutterSize matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: block;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        display: block;
        margin-bottom: 0.8rem;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-bottom: 0px;
      }
/>
`;

exports[`<LayoutList /> as vertical list without props matches its snapshot 1`] = `
<ul
  className=
      y-layoutList
      {
        display: block;
        list-style: none;
        margin-bottom: 0px;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item {
        display: block;
        margin-bottom: 0;
        margin-left: 0px;
        margin-right: 0px;
        margin-top: 0px;
        padding-bottom: 0px;
        padding-left: 0px;
        padding-right: 0px;
        padding-top: 0px;
      }
      & > .y-layoutList--item:last-child {
        margin-bottom: 0px;
      }
/>
`;
